internal/cmdparser/test.go

```diff
@@ -0,0 +1,60 @@
+package cmdparser
+
+import (
```

Comment:
the existence of both "test.go" and "test_test.go" is breaking my brain.

Reply:
I've added comments to both of these files.  test.go provides functions to write compact tests of the CLI application using this cmdparser package, and test_test.go tests these test function (testing the tests).  test_test.go is useful also, because it shows how easy it is to create a CLI using the cmdparser package.